Index: llvm-toolchain-snapshot_22~++20250908105315+a205695de4dc/clang/tools/libclang/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot_22~++20250908105315+a205695de4dc.orig/clang/tools/libclang/CMakeLists.txt
+++ llvm-toolchain-snapshot_22~++20250908105315+a205695de4dc/clang/tools/libclang/CMakeLists.txt
@@ -205,7 +205,7 @@ if(ENABLE_SHARED)
     endif()
     # Ensure that libclang.so gets rebuilt when the linker script changes.
     set_target_properties(libclang PROPERTIES
-                          VERSION ${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.${LLVM_VERSION_PATCH}${LLVM_VERSION_SUFFIX}
+                          VERSION ${LLVM_VERSION_MAJOR}
                           ${LIBCLANG_SOVERSION_ARG})
   endif()
 endif()
