Index: llvm-toolchain-snapshot_22~++20251018073733+f36c73e015a2/cmake/Modules/FindGRPC.cmake
===================================================================
--- llvm-toolchain-snapshot_22~++20251018073733+f36c73e015a2.orig/cmake/Modules/FindGRPC.cmake
+++ llvm-toolchain-snapshot_22~++20251018073733+f36c73e015a2/cmake/Modules/FindGRPC.cmake
@@ -49,10 +49,28 @@ else()
     message(FATAL_ERROR "gRPC C++ Plugin and Protoc must be on $PATH for gRPC-enabled build.")
   endif()
   if(NOT TARGET grpc++)
+    find_library(GPR_LIBRARY gpr ${GRPC_OPTS} REQUIRED)
+    add_library(gpr UNKNOWN IMPORTED GLOBAL)
+    message(STATUS "Using gpr: " ${GPR_LIBRARY})
+    set_target_properties(gpr PROPERTIES IMPORTED_LOCATION ${GPR_LIBRARY})
+
     find_library(GRPC_LIBRARY grpc++ REQUIRED)
     add_library(grpc++ UNKNOWN IMPORTED GLOBAL)
     message(STATUS "Using grpc++: " ${GRPC_LIBRARY})
     set_target_properties(grpc++ PROPERTIES IMPORTED_LOCATION ${GRPC_LIBRARY})
+
+    find_library(GRPC2_LIBRARY grpc ${GRPC_OPTS} REQUIRED)
+    add_library(grpc UNKNOWN IMPORTED GLOBAL)
+    message(STATUS "Using grpc: " ${GRPC2_LIBRARY})
+    set_target_properties(grpc PROPERTIES IMPORTED_LOCATION ${GRPC2_LIBRARY})
+
+    find_library(ABSL_SYNCHRONIZATION_LIBRARY absl_synchronization ${GRPC_OPTS} QUIET)
+    if (ABSL_SYNCHRONIZATION_LIBRARY)
+      add_library(absl_synchronization UNKNOWN IMPORTED GLOBAL)
+      message(STATUS "Using absl_synchronization: " ${ABSL_SYNCHRONIZATION_LIBRARY})
+      set_target_properties(absl_synchronization PROPERTIES IMPORTED_LOCATION ${ABSL_SYNCHRONIZATION_LIBRARY})
+    endif()
+
     if (ENABLE_GRPC_REFLECTION)
       find_library(GRPC_REFLECTION_LIBRARY grpc++_reflection REQUIRED)
       add_library(grpc++_reflection UNKNOWN IMPORTED GLOBAL)
